### PR TITLE
store events in event store upon events reception

### PR DESCRIPTION
**Pull request by AnthonyBuisset**
- 🚚 move event store append feature to event store crate
- ♻️ remove temporarly genericity of the event store
- ✨ append the received events in the event store
- 🔥 remove event store observer
- ♻️ publish event in contribution apply usecase
- ♻️ publish events in application refused usecase
- ♻ publish events in associate github account usecase
- ♻ publish events in register discord handle usecase
- 🐛 Fix event consuming when received in batches
- ♻️ move contributor refresh test to e2e
- ♻️ move project refresh tests to e2e
- ♻️ move contribution refresh tests to e2e
- ♻️ move applications refresh tests to e2e
- 🚨 fix clippy warnings

### CODE REVIEW

File: marketplace-event-store/src/lib.rs [hunk 1]
```diff
@@ -10,25 +11,35 @@ use anyhow::Result;
...
+		})
 		.await?;
 
 	Ok(())
 }
 
-async fn log(event: Event) -> Result<Event> {
-	let res_pretty_event = serde_json::to_string_pretty(&event);
-	if let Ok(pretty_event) = res_pretty_event {
+async fn log(store: Arc<dyn EventStore>, event: Event) -> Result<Event> {
```
Comment (Bernardstanislas) [inline marketplace-event-store/src/lib.rs:36]: I think that `log` does not represent what happens anymore. Maybe simply `store`?

File: marketplace-domain/src/event.rs [hunk 2]
```diff
@@ -11,6 +11,7 @@ pub enum Event {
 	Contributor(ContributorEvent),
 }
 
+// TODO: Remove after migration
```
Comment (Bernardstanislas) [inline marketplace-domain/src/event.rs:14]: 🙏

File: marketplace-indexer/src/lib.rs [hunk 3]
```diff
@@ -61,11 +61,6 @@ fn build_event_observer(
 	BlockchainObserverComposite::new(vec![
 		Arc::new(BlockchainLogger::default()),
 		Arc::new(event_store_bus),
-		Arc::new(EventStoreObserver::new(
```
Comment (Bernardstanislas) [inline marketplace-indexer/src/lib.rs:64]: image

File: marketplace-domain/src/messaging/publisher.rs [hunk 4]
```diff
@@ -13,6 +13,12 @@ pub enum Error {
 }
 
 #[async_trait]
-pub trait Publisher<M: Message>: Send + Sync {
+pub trait Publisher<M: Message + Sync>: Send + Sync {
 	async fn publish(&self, destination: Destination, message: &M) -> Result<(), Error>;
+	async fn publish_many(&self, destination: Destination, messages: &[M]) -> Result<(), Error> {
+		for message in messages {
+			self.publish(destination.clone(), message).await?;
```
Comment (Bernardstanislas) [inline marketplace-domain/src/messaging/publisher.rs:20]: Can we publish in parallel?

**Review by Bernardstanislas — APPROVED** [review]

File: marketplace-domain/src/messaging/publisher.rs (hunk 4 above)
Comment (AnthonyBuisset) [reply marketplace-domain/src/messaging/publisher.rs:20]: not if we want to keep the order

File: marketplace-event-store/src/lib.rs (hunk 1 above)
Comment (AnthonyBuisset) [reply marketplace-event-store/src/lib.rs]: done

**Pull request merged**